strats.md

```diff
@@ -45,3 +47,411 @@ Execution of a strat may have an effect of clearing one or more [obstacles](regi
+
+The `leaveWithSpark` object currently has no properties. If needed, properties might be added in the future to describe constraints on the position and direction of the spark. Currently it is implicitly assumed that the position can be fully controlled to be whatever is needed in the next room (e.g. to spark through either the top or bottom part of a horizontal door transition); so the requirements of a `leaveWithSpark` strat should be based on the worst-case scenario. The direction of the spark is assumed to be horizontal when sparking through horizontal door transitions, or vertical when sparking through vertical door transitions.
+
+*Note*: Using a runway connected to a door to leave the room with a shinespark is already covered by `leaveWithRunway`. Likewise `leaveCharged` implicitly includes the possibility of leaving the room with a shinespark. It is only necessary to use `leaveWithSpark` in cases where it would not be possible to reach the door before the shinecharge timer expires.
+
+#### Example
+```json
+{
+  "name": "Leave With Spark",
+  "notable": false,
+  "requires": [
+    {"canShinecharge": {
+      "usedTiles": 20,
+      "openEnd": 0
+    }},
+    {"shinespark": {
+      "frames": 200
+    }}
+  ],
+  "exitCondition": {
+    "leaveWithSpark": {}
+  }
+}
+```
+
+## Entrance conditions
+
+In all strats with an `entranceCondition`, the `from` node of the strat must be a door node. An `entranceCondition` object must contain exactly one property:
+
+- _comeInRunning_: This indicates that Samus must run into the room, with speed in a certain range.
+- _comeInJumping_: This indicates that Samus must run and jump just before hitting the transition, with speed in a certain range.
+- _comeInCharging_: This indicates that Samus must run into the room with enough space to complete a shinecharge.
+- _comeInCharged_: This indicates that Samus must enter the room with a shinecharge with a certain amount of frames remaining.
+- _comeInWithSpark_: This indicates that Samus must shinespark into the room.
+- _comeInWithBombBoost_: This indicates that Samus must come into the room with a horizontal bomb boost.
+- _comeInStutterCharging_: This indicates that Samus must run into the room with a stutter immediately before the transition.
+- _comeInWithDoorStuckSetup_: This indicates that Samus must enter the room in a way that allows getting stuck in the door as it closes.
+
+Each of these properties is described in more detail below.
+
+### Come In Running
+
+An `comeInRunning` entrance condition represents the need for Samus to be able to run into the room with speed in a certain range. It has the following properties:
+* _speedBooster_: If true, then Speed Booster must be used while running into the room. If false, then Speed Booster must not be used. If "any", then Speed Booster may or may not be used.
```

Comment:
if false, should that imply `canDisableEquipment`?

Reply:
I guess so? That could also go as a requirement in the strat though, so we wouldn't have to make it implicit. I could go either way. Might be worth checking first if there are any examples

Reply:
It would be better to need `canDisableEquipment` tech if speed needs to be off since that is the most confusing equipment item.  Those situations are harder to find though, probably short runways.  
I turned off speed for the tube gravity jump yesterday, when connected to a save room.  That would also have a crossroomJumpIntoWater tech on it.  Maybe the boulder room entry jump, or maybe not, for an example without a harder tech with it.
I expect most of the short runway jumps to need HiJump too and then it doesn't matter.

Reply:
Ok I added to the docs that there is an implicit `canDisableEquipment` when `speedBooster` is false.